pkg/stream/consumer_base.go

```diff
@@ -194,16 +233,61 @@ func (c *baseConsumer) runConsumerCallback(ctx context.Context) error {
 	return nil
 }
 
+func (c *baseConsumer) shovelData(ctx context.Context) error {
+	defer c.logger.Debug("shovelData is ending")
+	defer close(c.data)
+
+	cfn := coffin.New()
+	cfn.GoWithContextf(ctx, c.shovelDataFromSource(c.input, "input"), "panic during shoveling data from input")
+	cfn.GoWithContextf(ctx, c.shovelDataFromSource(c.retryHandler, "retry handler"), "panic during shoveling data from retry")
+
+	return cfn.Wait()
+}
+
+func (c *baseConsumer) shovelDataFromSource(input Input, src string) func(ctx context.Context) error {
+	return func(ctx context.Context) error {
+		defer c.logger.Debug("shovelDataFromSource %s is ending", src)
```

Comment:
Should we use the context here for the logger?

Reply:
the context shouldn't have any additional information here..